### PR TITLE
fix: store_aligned_words slice

### PR DESCRIPTION
**Pull request by enitrat**
## Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #270 

## What is the new behavior?

- Array is now moved 16 elements further instead of staying the same
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/memory.cairo [hunk 1]
```diff
@@ -262,7 +262,7 @@ impl InternalMemoryMethods of InternalMemoryTrait {
 
             self.items.insert(chunk_index.into(), current.try_into().unwrap());
             chunk_index += 1;
-            elements = elements.slice(0, 16);
+            elements = elements.slice(16, elements.len() - 16);
```
Comment (ClementWalter) [inline crates/evm/src/memory.cairo:265]: isn't this more expensive that just incrementing a start index in the loop?

**Review by ClementWalter — CHANGES_REQUESTED** [review]
isn't there any test to update? if no, a test to add I guess!

File: crates/evm/src/memory.cairo (hunk 1 above)
Comment (enitrat) [reply crates/evm/src/memory.cairo:265]: it's slightly better this way

File: crates/evm/src/tests/test_memory.cairo [hunk 2]
```diff
@@ -112,6 +112,58 @@ fn test_store_n_no_aligned_words() {
     assert(memory.bytes_len == 32, 'memory should be 32 bytes');
 }
 
+#[test]
+#[available_gas(200000000)]
+fn test_store_n_2_aligned_words() {
+    let mut memory = MemoryTrait::new();
+    let bytes_arr = array![
```
Comment (Eikix) [inline crates/evm/src/tests/test_memory.cairo:119]: should we have a util that inserts sequentially elements inside an array? 
pseudo-code:
```
fill_array_from_to(start: T, end: T) -> array<T>
```

might need to take oneable and add trait logic from herodotus

can be in a separate PR if you think its worth

Comment (enitrat) [reply crates/evm/src/tests/test_memory.cairo:119]: I don't see a usage in multiple places for now, let's open a PR later if it becomes required

**Comment by enitrat** [comment]
@ClementWalter can you re-approve pls

**Review by Eikix — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**